lightning/src/blinded_path/payment.rs

```diff
@@ -351,6 +357,22 @@ pub struct Bolt12OfferContext {
 	pub invoice_request: InvoiceRequestFields,
 }
 
+/// The context of a payment made for a static invoice requested from a BOLT 12 [`Offer`].
+///
+/// [`Offer`]: crate::offers::offer::Offer
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct AsyncBolt12OfferContext {
+	/// The identifier of the [`Offer`].
+	///
+	/// [`Offer`]: crate::offers::offer::Offer
+	pub offer_id: OfferId,
```

Comment:
This is used to fill in the `Bolt12OfferContext::offer_id` field. I ended up opening the follow-up if you want to check out more context: https://github.com/lightningdevkit/rust-lightning/pull/3440

Reply:
This `offer_id` is in the static invoice for its blinded payment path contexts, not the offer

Reply:
Ah, that's true, it looks like we actually already re-calculate the `offer_id` when calling `InvoiceRequest::verify_using_recipient_data`. I think it might not be exposed currently but should be able to add an accessor